### PR TITLE
MAT-21 and MAT-22 settings page components and state management

### PR DESCRIPTION
**Pull request by sheros21**

### CODE REVIEW

File: src/Pages/Settings.tsx [hunk 1]
```diff
@@ -1,10 +1,380 @@
...
+  const frontload: CreateUsersModelInput = {
+    user_name: '',
+    email: '',
+    first_name: '',
+    last_name: '',
+    password: ''
+  }
+  const finalload: CreateUsersPayload = {
+    input: frontload
+  }
```
Comment (akumar282) [inline src/Pages/Settings.tsx:58]: Why is create user functionality in settings

**Comment by haylau** [comment]
builds and passes jest tests 👍

File: src/Pages/Settings.tsx [hunk 2]
```diff
@@ -1,10 +1,380 @@
...
+        validationSchema: validationSchema,
+        onSubmit: (values) => {
+          // Link to preferences page
+          // alert(JSON.stringify(values, null, 2));
+          sendToDatabase(values)
+        },
+        
+        ///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+        ///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+        ///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
```
Comment (akumar282) [inline src/Pages/Settings.tsx:182]: Why?

File: src/Pages/Settings.tsx [hunk 3]
```diff
@@ -1,10 +1,380 @@
...
+  const handleChange = (event: any, newValue: any | null) => {
+    setSelectedTags(newValue.map(option => option.value || option));
+    console.log(typeof(selectedTags));
+    console.log(selectedTags);
+    console.log("change");
+  };
+  
+    ///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+    ///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+    ///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
```
Comment (akumar282) [inline src/Pages/Settings.tsx:110]: Why again?

**Review by haylau — APPROVED** [review]
should do:
- cleanup whitespace

probably should do/up to you:
- remove comment dividers 
- function docs to block comments 

otherwise LGTM

**Review by akumar282 — CHANGES_REQUESTED** [review]

**Comment by akumar282** [comment]
I think that the TODO items that hayden mentioned are needed

File: src/Pages/Settings.tsx (hunk 1 above)
Comment (haylau) [reply src/Pages/Settings.tsx:58]: if it's change password/ etc. we still need to write specific mutations on the backend to modify vs create

**Comment by mohazahid** [comment]
I would add a button under filter settings that causes a submit so we wouldn't have to worry about checking for changes.
One of the passwords has a spelling mistake
Also I would always make sure to git pull origin main before pr so that your branch is up to date with current main changes
Otherwise LGTM
+what abhi said about comment blocks

File: src/Pages/Settings.tsx (hunk 1 above)
Comment (sheros21) [reply src/Pages/Settings.tsx:58]: The code block is just to reference later for connecting the settings page to the backend, it can be ignored for now

**Comment by mohazahid** [comment]
Also for future reference once backend is setup set the initial states in formik to the values before edit instead of placeholders because formik will think of a placeholder as not really there and will try to submit an empty string

**Comment by akumar282** [comment]
That code block is still wrong since settings would be of type update user. You should be able to find the correct type in `src/backend/types.ts`

**Comment by sheros21** [comment]
> That code block is still wrong since settings would be of type update user. You should be able to find the correct type in `src/backend/types.ts` Mohammads the best

As mentioned before, the code block is for future reference. A connection to the backend will be made on a separate branch for a different ticket.

**Comment by sheros21** [comment]
> That code block is still wrong since settings would be of type update user. You should be able to find the correct type in `src/backend/types.ts` Mohammads the best

As mentioned before, the code block is for future reference. A connection to the backend will be made on a separate branch for a different ticket

**Pull request merged**